codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/SyncClientInterface.java

```diff
@@ -97,8 +97,8 @@ public TypeSpec poetSpec() {
                                               ServiceMetadataProvider.class)
                                   .build());
 
-        Optional<ClassName> extensionClass = findExtensionInterface(className, model.getCustomizationConfig());
-        extensionClass.ifPresent(result::addSuperinterface);
+        Optional<ClassName> extensionInterface = findExtensionInterface(className, model.getCustomizationConfig());
```

Comment:
The overloaded meaning of class in Java makes naming difficult sometimes

Reply:
Very true, this is still a "class" by most definitions.